pkg/platform/openstack/kuryr_bootstrap.go

```diff
@@ -331,18 +334,56 @@ func BootstrapKuryr(conf *operv1.NetworkSpec, kubeClient client.Client) (*bootst
 		return nil, errors.Wrap(err, "failed to get cloud provider CA certificate")
 	}
 
+	// We cannot rely on the inject-proxy annotation because the CVO, which is
+	// responsible to inject the proxy env vars, is not available before CNO.
+	proxyConfig := &configv1.Proxy{}
+	err = kubeClient.Get(context.TODO(), types.NamespacedName{Name: names.CLUSTER_CONFIG}, proxyConfig)
+	if err != nil && !apierrors.IsNotFound(err) {
+		return nil, err
+	}
+
+	transport := http.Transport{}
+	noProxy := proxyConfig.Status.NoProxy
+	httpProxy := proxyConfig.Status.HTTPProxy
+	httpsProxy := proxyConfig.Status.HTTPSProxy
+	hasProxy := len(httpsProxy) > 0 || len(httpProxy) > 0 || len(noProxy) > 0
+	if hasProxy {
+		os.Setenv("NO_PROXY", noProxy)
+		os.Setenv("HTTP_PROXY", httpProxy)
+		os.Setenv("HTTPS_PROXY", httpsProxy)
+		// The env vars are not propagated to different libs when not set on
+		// main(), so we'll load it directly here and rely on http lib to choose
+		// the proxy URL.
+		proxyfunc := httpproxy.FromEnvironment().ProxyFunc()
+		transport.Proxy = func(req *http.Request) (*url.URL, error) {
+			return proxyfunc(req.URL)
+		}
+		provider.HTTPClient = http.Client{Transport: &transport}
+
+		// Due to an issue in the urllib3 library https://github.com/psf/requests/issues/5939
+		// Kuryr will always default to use http scheme when https is set.
+		proxyUrl, err := url.Parse(httpsProxy)
+		if err != nil {
+			return nil, errors.Wrap(err, "failed to parse cluster-wide proxy https URL")
+		}
+
+		if proxyUrl.Scheme == "https" {
+			log.Printf("Kuryr only supports cluster-wide http scheme.")
+			if len(httpProxy) > 0 {
+				log.Printf("Defaulting cluster-wide proxy to %s", httpProxy)
+				httpsProxy = httpProxy
```

Comment:
`HTTP_PROXY` can be `https` too... in fact, `HTTP_PROXY` and `HTTPS_PROXY` are almost always going to be exactly the same, so this check is probably useless.

Reply:
It seems CNO is only allowing `http` scheme at `HTTP_PROXY`
https://github.com/openshift/cluster-network-operator/blob/master/pkg/controller/proxyconfig/validation.go#L42-L47

Reply:
And the validation of install-config produces the following:
`"Install Config": invalid "install-config.yaml" file: proxy.httpProxy: Unsupported value: "https": supported values: "http"`

Reply:
huh, ok, that's interesting. In that case, I guess this code is reasonable